README.md

```diff
@@ -134,6 +134,79 @@ into their own mount. This could be achieved with a configuration like:
                 - secret
 
 
+### Use global `:hierarchy` - optional
+By default, only the list of mounts is traversed as described above.
+However, when having configured:
+
+    :vault:
+        :use_hierarchy: true
+
+the `:hierarchy` source paths from the hiera configuration are used on top of each mount.
+This makes the behavior of the vault backend the same as other backends.
+Additionally, this enables usage of the third parameter to the hiera functions in puppet,
+the so-called 'override' parameter.
+
+Example: In case we have the following hiera config:
+
+    :backends:
+        - vault
+        - yaml
+
+    :hierarchy:
+      - nodes/%{fqdn}
+      - hostclass/%{hostclass}
+      - location/%{location}
+      - zone/%{zone}
+      - companyenv/%{companyenv}
+      - ostype/%{operatingsystem}
+      - ostype/%{operatingsystem}-%{lsbdistcodename}
+      - defaults
+
+    :yaml:
+      :datadir: /var/lib/hiera/%{environment}/
+
+    :vault:
+        :addr: ...
+        :use_hierarchy: true
+        :mounts:
+            :generic:
+                - %{environment}
+                - secret
+
+Each lookup will now for each mount loop through each path from the `:hierarchy`
+list, on top of the mount. So, the paths it will look for entries are:
+
+    %{environment}/nodes/%{fqdn}
+    %{environment}/hostclass/%{hostclass}
+    ...
+    %{environment}/defaults
+    secret/nodes/%{fqdn}
+    secret/hostclass/%{hostclass}
+    ...
+    secret/defaults
+
+Additionally, the third argument to the hiera functions, the override parameter is supported.
```

Comment:
It might be worth linking to the hiera docs in this section: http://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions